compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -426,6 +427,89 @@ static LogicalResult setUserConfig(
   return success();
 }
 
+/// Return the size of the given dimension in the linalg op.
+// TODO: this should be part of LinalgOp interface, the equivalent member
+// function currently only support the case where all the dimensions are static
+// while we want to support dynamic shapes.
+static Optional<int64_t> getLinalgDimSize(linalg::LinalgOp op, int64_t d) {
+  for (auto map : llvm::enumerate(op.getIndexingMaps())) {
+    for (auto dim : llvm::enumerate(map.value().getResults())) {
+      if (dim.value().cast<AffineDimExpr>().getPosition() == d) {
```

Comment:
Could we change this to use `dyn_cast` instead of `cast` and return llvm::None on failure.